### PR TITLE
RAI-11434 refactor polling transaction logic

### PR DESCRIPTION
**Pull request by OsamaSBCrea**

### CODE REVIEW

File: src/rest.ts [hunk 1]
```diff
@@ -33,6 +33,22 @@ export type RequestOptions = {
...
 };
 
+export type PollOptions = {
+  overheadRate?: number;
+  startTime?: number;
+  maxInterval?: number;
+  timeout?: number;
+};
+
+export type PollingResult<T> =
```
Comment (denisgursky) [inline src/rest.ts:43]: Hm, why not just
```
export type PollingResult<T> = {
  done: boolean;
  result?: T;
};
```

File: src/rest.ts [hunk 2]
```diff
@@ -33,6 +33,22 @@ export type RequestOptions = {
   onResponse?: (r: ApiResponse) => void;
 };
 
+export type PollOptions = {
+  overheadRate?: number;
+  startTime?: number;
```
Comment (denisgursky) [inline src/rest.ts:38]: `startTime` looks kinda weird, doesn't it?

File: src/api/query/execAsyncApi.ts [hunk 3]
```diff
@@ -61,9 +60,9 @@ export class ExecAsyncApi extends TransactionAsyncApi {
     inputs: QueryInput[] = [],
     readonly = true,
     tags: string[] = [],
-    interval = 1000, // 1 second
     timeout = Number.POSITIVE_INFINITY,
```
Comment (denisgursky) [inline src/api/query/execAsyncApi.ts:63]: Should be `options?: PollOptions,` to be consistent with the poll function

File: src/rest.ts (hunk 2 above)
Comment (OsamaSBCrea) [reply src/rest.ts:38]: So do we want to hardcode it in the `pollWithOverhead` function to `Date.now`, or we should just not expose it in the `exec` and `pollTransaction` options?

Comment (denisgursky) [reply src/rest.ts:38]: Hardcode in `pollWithOverhead`

**Review by denisgursky — APPROVED** [review]

**Pull request merged**